src/components/calculator.js

```diff
@@ -1,51 +1,118 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 
 export default class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: 0,
+      next: 0,
+      operation: null,
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleClick = this.handleClick.bind(this);
   }
 
+  handleChange(e) {
+    this.setState({ next: e.target.value });
+  }
+
+  handleClick(e) {
+    if (this.isNumber(e.target.outerText)) {
+      console.log('num');
+      this.setState({ next: e.target.outerText });
+      const obj = this.state;
+      const res = calculate(obj, e.target.outerText);
+      this.setState({ next: res.next });
+      console.log(res);
+    }
+
+    if (e.target.outerText === '+' || e.target.outerText === '-' || e.target.outerText === 'x' || e.target.outerText === '÷') {
+      this.setState({ operation: e.target.outerText });
+      const obj = this.state;
+      const res = calculate(obj, e.target.outerText);
+      this.setState({ next: res.next });
+      this.setState({ total: res.total });
+      console.log(res);
+      console.log(this.state);
+    }
+
+    if (e.target.outerText === '=') {
+      const obj = this.state;
+      const res = calculate(obj, e.target.outerText);
+      this.setState({ ...res });
+      this.setState({ next: res.total });
+      console.log(res);
+      console.log(this.state);
+    }
+    if (e.target.outerText === '.') {
+      const obj = this.state;
+      const res = calculate(obj, e.target.outerText);
+      console.log(res);
+      if (Object.entries(res).length !== 0) {
+        this.setState({ ...res });
+        this.setState({ next: res.next });
+      }
+    }
+    if (e.target.outerText === 'AC') {
+      const reset = { total: 0, next: 0, operation: null };
+      this.setState({ ...reset });
+    }
+    if (e.target.outerText === '+/-') {
+      const obj = this.state;
+      const res = calculate(obj, e.target.outerText);
+      this.setState({ ...res });
+      this.setState({ next: res.next });
+    }
+    if (e.target.outerText === '%') {
+      const obj = this.state;
+      this.setState({ next: parseFloat(obj.next / 100).toString() });
+    }
+  }
+
+  isNumber=(item) => !!item.match(/[0-9]+/)
+
   render() {
+    const { next } = this.state;
     return (
       <div className="calculator">
         <div className="calculator-body">
           <div className="input-val">
-            <input type="text" value="0" />
+            <input type="text" value={next} onChange={this.handleChange} />
```

Comment:
* I get errors in the console when i click on operation buttons (ie +, -) 
![image](https://user-images.githubusercontent.com/52497246/129136559-4d760376-6139-4868-b391-aa6ef41c784e.png)